test/support/sequence_coverage_reporting.rb

```diff
@@ -60,49 +65,141 @@ def add_assertion_location(location)
+# indirect assertion calls. Direct assertion calls are assertions that are
+# called directly in a test or method being tested. Indirect assertion calls are
+# assertions that are called within a method that is called by the test being
+# tested.
+#
+# test '1' do
+#   assert true # direct assertion call
+# end
+#
+# test '2' do
+#   validate
+# end
+#
+# def validate
+#   assert true # Direct assertion call if we are unit testing the validate
+#               # method.
+#               # Indirect assertion call if we are unit testing test '2'.
+# end
 class AssertionReporter
   class << self
     def report
       create_csv
-      print_unknown_call_sites
     end
 
     def create_csv
       CSV.open(File.join(__dir__, '..', '..', 'sequence_coverage.csv'), 'wb') do |csv|
-        csv << ['Assertion Location', 'Pass Count', 'Fail Count']
+        csv << [
+          'Assertion Location',
+          'Direct Pass Count',
+          'Direct Fail Count',
+          'Indirect Pass Count',
+          'Indirect Fail Count'
+        ]
         AssertionTracker.assertion_locations.sort.each do |location|
-          results = AssertionTracker.assertion_calls[location]
-          csv << [location, pass_count(results), fail_count(results)]
+          csv << [
+            location,
+            direct_pass_count(location),
+            direct_fail_count(location),
+            indirect_pass_count(location),
+            indirect_fail_count(location)
+          ]
         end
       end
     end
 
-    def pass_count(results)
-      results.count { |result| result }
+    def direct_call?(location)
+      location.include? AssertionCallLocationFormatter::CALL_SEPARATOR
     end
 
-    def fail_count(results)
-      results.count(&:!)
+    def direct_pass_count(location)
+      AssertionTracker.assertion_calls[location].count { |result| result }
     end
 
-    # Locations which called an assertion which was not detected in the AST.
-    # This happens when a method from outside of the sequence makes an assertion
-    # (e.g., a method from SequenceBase).
-    def unknown_call_sites
-      AssertionTracker.assertion_calls.keys - AssertionTracker.assertion_locations
+    def direct_fail_count(location)
+      AssertionTracker.assertion_calls[location].count(&:!)
     end
 
-    def print_unknown_call_sites
-      return if unknown_call_sites.blank?
+    def indirect_pass_count(location)
+      return if direct_call? location
 
-      puts "\nUnrecognized assertion call sites:"
-      puts unknown_call_sites.join("\n")
+      AssertionTracker.assertion_calls.reduce(0) do |count, (key, results)|
+        key.end_with?(location) && key != location ? count + results.count { |result| result } : count
+      end
+    end
+
+    def indirect_fail_count(location)
+      return if direct_call? location
+
+      AssertionTracker.assertion_calls.reduce(0) do |count, (key, results)|
+        key.end_with?(location) && key != location ? count + results.count(&:!) : count
```

Comment:
This line can be a little tricky to read with all the operations going on. I don't think it has to change, but even moving `key.end_with?(location) && key != location` out might make it more clear, although at the expense of another line.